### PR TITLE
refact: keccak hashing func using std crypto

### PR DESCRIPTION
**Pull request by StringNick**

### CODE REVIEW

**Comment by tcoratger** [comment]
@StringNick I like it, much better like that, tell me when ready to review/merge as it is marked as draft now but seems already good with all the tests passing :)

**Comment by StringNick** [comment]
> @StringNick I like it, much better like that, tell me when ready to review/merge as it is marked as draft now but seems already good with all the tests passing :)

u can review, i pushed for Lana review, if u can add comments will be cool, i didnt remove old code, u can review new code i will remove after (old code)

File: src/vm/builtins/builtin_runner/keccak.zig [hunk 1]
```diff
@@ -265,53 +265,42 @@ pub const KeccakBuiltinRunner = struct {
...
-            @as(
-                *[KeccakPrimitives.PLEN]u64,
-                @ptrCast(vec.items.ptr),
-            ),
-        ) |item| {
-            var buf: [8]u8 = undefined;
-            std.mem.writeInt(u64, buf[0..], item, .little);
-            try result.appendSlice(&buf);
+        for (st.st, 0..) |item, idx| {
+            std.mem.writeInt(u64, result[idx * @sizeOf(u64) .. (idx + 1) * @sizeOf(u64)][0..@sizeOf(u64)], item, .little);
```
Comment (tcoratger) [inline src/vm/builtins/builtin_runner/keccak.zig:303]: Here as there are lots of arguments inside the `writeInt`, maybe put a comma at the end for multi lines fmt.

File: src/vm/builtins/builtin_runner/keccak.zig [hunk 2]
```diff
@@ -265,53 +265,42 @@ pub const KeccakBuiltinRunner = struct {
...
-        var result = ArrayList(u8).init(allocator);
-        var vec = ArrayList(u64).init(allocator);
-        defer vec.deinit();
+    /// An fixed array based on block type, containing the Keccak hash.
+    /// current block type hardcoded to u64.
+    fn keccakF(input_message: *[]const u8) ![KeccakPrimitives.PLEN * @sizeOf(u64)]u8 {
+        var result = [_]u8{0} ** (KeccakPrimitives.PLEN * @sizeOf(u64));
+
+        // 1600 bits = 200 bytes = 25 u64
+        var st: std.crypto.core.keccak.KeccakF(@bitSizeOf(u64) * KeccakPrimitives.PLEN) = .{
```
Comment (tcoratger) [inline src/vm/builtins/builtin_runner/keccak.zig:283]: Maybe we can find another name for this variable, in order to avoid `st.st` usage after that, wdt?

Comment (StringNick) [reply src/vm/builtins/builtin_runner/keccak.zig:283]: st -> hashState?

Comment (tcoratger) [reply src/vm/builtins/builtin_runner/keccak.zig:283]: Yes sounds good

**Comment by StringNick** [comment]
@tcoratger pushed, removed old code, moved constant round count

**Review by tcoratger — APPROVED** [review]

**Pull request merged**